service/frontend/nexus_handler.go

```diff
@@ -427,18 +432,19 @@ func (h *nexusHandler) StartOperation(
 		return nil, nexus.HandlerErrorf(nexus.HandlerErrorTypeBadRequest, "input exceeds size limit")
 	}
 
+	if deadline, hasDeadline := ctx.Deadline(); hasDeadline && time.Until(deadline) < nexusDispatchPreflightTimeout {
```

Comment:
Awesome, there are some SDK tests that might fail because of this. We would need to modify those or make this configurable. Not blocking the PR as this is not technically meant for users to invoke directly.